cmd/agent/app/flare.go

```diff
@@ -79,6 +92,16 @@ var flareCmd = &cobra.Command{
 }
 
 func readProfileData(pdata *flare.ProfileData) error {
+	c, err := common.NewSettingsClient()
+	if err != nil {
+		return fmt.Errorf("failed to initialize settings client: %v", err)
+	}
+	prevSettings := make(map[string]interface{})
+	if err := setRuntimeProfilingSettings(c, &prevSettings); err != nil {
```

Comment:
Did you consider passing `runtime_block_profile_rate` and `runtime_block_profile_rate` as parameters to the api call like `profiling` (The api handler can save the previous state and restore it when done) 
One of the advantage is that you can curl the url `http://127.0.0.1:%s/debug/pprof/...` without manually restore the profile rates.

There is no need to change anything, I am just curious if it would be possible.

Reply:
Entire `/debug/pprof` is provided as-is by the pprof package, including the `profile` endpoint we call to do the profiling. We could override it, of course, that would be bit surprising. I don't think a curl user would expect the endpoint to have additional undocumented parameters.